org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/GradleBuild.java

```diff
@@ -84,4 +88,36 @@
      * @return the request attributes, never null
      */
     FixedRequestAttributes getRequestAttributes();
+
+    /**
+     * Creates a new Gradle build invocation.
+     *
+     * @param transientAttributes the transient attributes for the invocation
+     * @param config the configuration for the build launcher
+     * @return the object that can run the Gradle build
+     */
+    GradleInvocation newBuildInvocation(TransientRequestAttributes transientAttributes, BuildLauncherConfig config);
```

Comment:
We should not introduce new types here. Instead we should just return a `TestLauncher` (the interface from the TAPI).

The underlying project connection can be closed transparently with a progress listener that the `DefaultGradleBuild` registers. When the "build finished" event is received, we can call `ProjectConnection.close()`.